### ISSUE TITLE
No module named production

### ISSUE THREAD

**Issue by travlr** [regisb/tutor#202]
In general, what might be causing this problem?

## Steps to reproduce

`tutor local quickstart`

* Using custom tutor with custom template settings.
  * lms.env.json / cms.env.json
  * private.txt
* Using my own openedx:ironwood.1 image without any customization.

## Unexpected behavior

![Screenshot_20190419_115021](https://user-images.githubusercontent.com/125945/56436535-683f2d80-6299-11e9-96e0-57b1b5a63c7d.png)

## Additional info (IMPORTANT)

I am researching what I may have done to cause this issue.. I'll report back

Provide the output of `tutor --version`.

* tutor version is master with some custom template settings.

**Comment by regisb** [comment]
Hi @travlr! Which version of edx-platform are you running? You can check the git version with:

    tutor local run lms git log -1

**Comment by travlr** [comment]
Hi @regisb 

tutor, version 3.3.4

I reverted to using just regisb/tutor:master and that worked.. so, I'm checking for the change I made that caused the issue. 

Which brings me to the question: Is the case that inorder preserve the `env/` directory, I must place my lms.env.json and cms.env.json in the tutor templates (respectively)? Is this the way I should do `requirements/private.txt`?

My purpose for all of this, is to install custom XBlocks.

**Comment by travlr** [comment]
prodatalab@ubuntu-s-4vcpu-8gb-nyc3-01:~/dev/src/github.com/first-contact-crypto/tutor$ tutor local run lms git log -1
docker-compose -f /home/prodatalab/.local/share/tutor/env/local/docker-compose.yml --project-name tutor_local run --rm lms git log -1
Starting tutorlocal_memcached_1 ... 
Starting tutorlocal_mysql_1 ... . . done
Starting tutorlocal_elasticsearch_1 ... done
Starting tutorlocal_forum_1 ... done
commit 4241edf726a903671a92642e6ae63ae84ec99f68
Author: Ned Batchelder <ned@nedbatchelder.com>
Date:   Sat Mar 16 06:09:38 2019 -0400

    Merge pull request #20007 from edx/nedbat/ironwood-19845
    
    Make the studio login over the lms optional using a feature flag

**Comment by regisb** [comment]
Peter, since the issue appears to be resolved by the current version of Tutor, I'll close this issue, but feel free to comment on it.

> Is the case that inorder preserve the env/ directory, I must place my lms.env.json and cms.env.json in the tutor templates (respectively)? Is this the way I should do requirements/private.txt?

The `env` directory is overwritten every time you run `tutor config save` (and thus `tutor local quickstart`). In order to make arbitrary modifications to the rendered templates, you need to modify the raw templates in `tutor/templates`.

> My purpose for all of this, is to install additional XBlocks.

To do so, you might not need to modify the tutor templates. Are you aware of https://docs.tutor.overhang.io/customise.html#installing-extra-xblocks-and-requirements ?

**Comment by travlr** [comment]
Sorry, the {lms,cms}.env.json changes were general config changes except for xblock-badgr has some XBlock settings that need to be in there. Yes, thank you I am familiar with that documentation.. I'm just trying to avoid my env being refreshed everytime I might do a `tutor local quickstart` or a `tutor config save`.. If consolidating these configs outside of the tutor binary is on your roadmap.. I can try to work on that. I know that there is a PR in regards to this issue.. to a degree.

**Comment by regisb** [comment]
Ok, I understand. For now, you should avoid doing `tutor config save` or `tutor local quickstart`. Instead, you should rely on the individual `tutor local` commands, such as `tutor local databases` and `tutor local https create`. All commands called by `quickstart` are here: https://github.com/regisb/tutor/blob/master/tutor/local.py#L26

Of course, I want to avoid to maintain small patches for all the xblocks in the world; at the same time, I also want to avoid people forking tutor just to support their specific requirements, such as xblock settings. And I *also* want to support the compatibility of tutor with all the xblocks in the world... To achieve these three objectives, I will implement a plugin system for tutor: tutor developers will be able to patch their deployments by creating simple plugins. This will make it possible for anyone to distribute their xblock, their app, or their battle-hardened security configuration to the rest of the Open edX community.

tl;dr: for now, you have to maintain your own fork of Tutor, but soon you will be able to create to run the stable release with custom plugins :-)
